src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt

```diff
@@ -0,0 +1,37 @@
+package Properties.Tamir
+import java.time.DayOfWeek
+
+//1. create class called Store that initlize by day and list of products
+data class Store(val day: DayOfWeek,val products: List<Product>)
+{
+    //2. add property to that indicate if the store is open the store is open all the day expect saturday
+    val shomeretShabat:Boolean get() {
+        when(day){
+            DayOfWeek.SATURDAY -> return false
+            else -> return true
+        }
+    }
+
+    //3. add property to that indicate number of product
+    val numberOfProducts:Int get() {
+        return products.size
+    }
+
+    //5. I want to count number of calling get receipts
+    var countGetReceipts:Int = 0
+
+    //4. add val that get receipts //no need to implement the method but its an heavy task
+    val getReceipts:List<String> by lazy {
```

Comment:
using lazy here is not will do the trick, 
I'l explain in the first time you call getReceipts you will increase the counter but you also going to set the value to listOf("Hello","Bob"). meaning in the next time you going to call to getReceipts you';; get the list witout increase the counter.
In order to solve it you can use get() that store function and not value.

Reply:
Got it 
thanks